demisto_sdk/commands/changelog/changelog.py

```diff
@@ -0,0 +1,251 @@
+                )
+                return False
+            if not self.validate_log_yml():
+                return False
+
+        return True
+
+    """ INIT """
+
+    def init(self, breaking: str = None, feature: str = None, fix: str = None) -> None:
+        """
+        Creates a new log file for the current PR
+        - if `fix` or `breaking` or `feature` is provided
+          it will create the file with the provided arguments
+        - if no argument is provided, it will create the file with initial values,
+          then the user has to change the values manually
+        """
+        if self.is_release():
+            logger.error(
+                "This PR is for release, please use in changelog release command"
+            )
+
+        msg = (
+            f"The creation of the log file .changelog/{self.pr_number}.yml is complete"
+        )
+
+        if breaking or feature or fix:
+            log = self.create_log(breaking, feature, fix)
+        else:
+            msg += ",\nPlease go to the file and edit the initial values."
+            log = INITIAL_LOG
+
+        with (CHANGELOG_FOLDER / f"{self.pr_number}.yml").open("w") as f:
+            yaml.dump(log, f)
+
+        logger.info(msg)
+
+    """ RELEASE """
+
+    def release(self) -> None:
+        if not self.is_release():
+            logger.error("The name of the PR is not valid for a release")
+            raise ValueError()
+        logs = self.get_all_logs()
+        self.extract_and_build_changelogs(logs)
+        self.cleaning_changelogs_folder()
+        logger.info("The build of the changelog for the release is complete")
+
+    """ HELPER FUNCTIONS """
+
+    def is_changelog_changed(self) -> bool:
+        return (
+            "CHANGELOG.md"
+            in Repo(".").git.diff("HEAD..origin/master", name_only=True).split()
+        )
+
+    def is_log_folder_empty(self) -> bool:
+        return not any(CHANGELOG_FOLDER.iterdir())
+
+    def is_log_yml_exist(self) -> bool:
+        return (CHANGELOG_FOLDER / f"{self.pr_number}.yml").is_file()
+
+    def validate_log_yml(self) -> bool:
+        data = get_yaml(CHANGELOG_FOLDER / f"{self.pr_number}.yml")
+
+        try:
+            LogObject(**data)
+        except ValidationError as e:
+            logger.error(e.json())
+            return False
+        return True
+
+    def get_all_logs(self) -> List[LogObject]:
+        changelogs: List[LogObject] = []
+        for path in CHANGELOG_FOLDER.iterdir():
+            changelog_data = get_yaml(path)
+            try:
+                changelogs.append(LogObject(**changelog_data))
+            except ValidationError as e:
+                self.handle_error += f"{path}: {e.json()}\n"
+
+        if self.handle_error:
+            raise ValueError(
+                f"The following log files are invalid:ֿֿ\n{self.handle_error}"
+            )
+        return changelogs
+
+    def extract_and_build_changelogs(self, logs: List[LogObject]) -> None:
```

Comment:
instead of modifying `self` (harder to maintain and follow the code), please return a value.